### PR TITLE
[ISSUE-101] Drive Replacement proposal

### PR DESCRIPTION
**Pull request by mishoyama**
## Purpose
[ISSUE-101](https://github.com/dell/csi-baremetal/issues/96) Proposal for drive replacement

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved
- [x] PR validation is passed
- [ ] Custom CI is passed

## Testing
Not relevant

### CODE REVIEW

File: docs/proposals/drive-replacement.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+- `OPERATIVE` - drive in use
+- `RELEASING` - releasing in progress
+- `RELEASED` - drive ready for removal
+- `FAILED` - drive failed to remove
+- `REMOVING` - removing in progress (PVs deletion, Safe remove, LED locate)
+- `REMOVED` - drive removed
+### API
+#### Operator
+Negotiation between CSI and Operator is optional and will be done though annotation on different objects.
+* Storage class annotation:
```
Comment (rususlasan) [inline docs/proposals/drive-replacement.md:54]: Storage class it is a cluster level component which could be used by different operators. Does it mean that all who works with such SC must support volume release feature?

Comment (mishoyama) [reply docs/proposals/drive-replacement.md:54]: This annotation must be set for PVC. I'll update this part.

Comment (mishoyama) [reply docs/proposals/drive-replacement.md]: Updated.

**Review by rususlasan — APPROVED** [review]

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Review by ykulazhenkov — APPROVED** [review]

File: docs/proposals/drive-replacement.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+* If recovery failed due to some reason Operator must put `volumerelease.csi-baremetal/release: failed` annotation on volume CR(s)
+  - In addition Operator can provide additional information about error `volumerelease.csi-baremetal/status: <status description>`
+  - CSI sets drive operational status to `FAILED`
+* When drive operation status is `RELEASED` user can initiate physical drive replacement by setting `driveremove.csi-baremetal/replacement: ready` annotation on drive CR
+  - CSI sets drive operational status to `REMOVING`
+  - Operator deletes PV(s)
+  - CSI prepares drive for safe removal and starts LED locate
+  - CSI sets drive operational status to `REMOVED` if all operations are passed successfully and `FAILED` otherwise
+* When drive operation status is `REMOVED`
+  - Use does physical replacement
```
[the remark this answers is not in the document]
Comment (mishoyama) [reply docs/proposals/drive-replacement.md]: Updated

**Pull request merged**